HPXMLtoOpenStudio/resources/weather.rb

```diff
@@ -251,6 +252,35 @@ def calc_ashrae_622_wsf(rowdata, epw_file)
     return wsf.round(2)
   end
 
+  # TODO
+  #
+  # @param epw_file [OpenStudio::EpwFile] OpenStudio EpwFile object
+  # @return [TODO] TODO
+  def get_header_info_from_epw(epw_file)
+    header.City = epw_file.city
```

Comment:
This is what allows purging the `epw_file` object. We store any relevant information from that object on the weather object, so that the weather object has *everything*.